File: starter-code/python_example_telemetry_api_extension/extensions_api_client.py
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: MIT-0

import os
import sys
import requests
import json


REGISTRATION_REQUEST_BASE_URL = "???"

def register_extension(extension_name):
    print ("[extension_api_client.register_extension] Registering Extension using {0}".format(REGISTRATION_REQUEST_BASE_URL))
    
    try:
        #OUR CODE GOES HERE

    except Exception as e:
        print("[extension_api_client.register_extension] Error registering extension: ",e, flush=True)
        raise Exception("Error setting AWS_LAMBDA_RUNTIME_API", e)


def next(extension_id):

    try:
        #OUR CODE GOES HERE

    except Exception as e:
        print("[extension_api_client.next] Error registering extension.", e, flush=True)
        raise Exception("Error setting AWS_LAMBDA_RUNTIME_API", e)

File: starter-code/nodejs_example_telemetry_api_extension/nodejs-example-telemetry-api-extension/extensions_api_client.js
const fetch = require('node-fetch');
const {basename} = require('path');

const REGISTRATION_REQUEST_BASE_URL = `http://${process.env.AWS_LAMBDA_RUNTIME_API}/2020-01-01/extension`;

async function register() {
    console.info('[extensions-api:register] Registering using REGISTRATION_REQUEST_BASE_URL', REGISTRATION_REQUEST_BASE_URL);

    //OUR CODE GOES HERE

}

async function next(extensionId) {
    console.info('[extensions-api:next] Waiting for next event');

    //OUR CODE GOES HERE

}

module.exports = {
    register,
    next,
};